engine/runtime-compiler/src/test/java/org/enso/compiler/test/mock/VirtualFileSystem.java

```diff
@@ -0,0 +1,212 @@
+    return parent.resolve(childName);
+  }
+
+  @Override
+  public Path getParent(Path path) {
+    return path.getParent();
+  }
+
+  @Override
+  public boolean exists(Path file) {
+    return Files.exists(file);
+  }
+
+  @Override
+  public void createDirectories(Path file) throws IOException {
+    Files.createDirectories(file);
+  }
+
+  @Override
+  public Path relativize(Path parent, Path child) {
+    return parent.relativize(child);
+  }
+
+  @Override
+  public Iterable<String> getSegments(Path file) {
+    var absPath = file.toString();
+    return Arrays.stream(absPath.split(fileSystem.getSeparator()))
+        .dropWhile(String::isEmpty)
+        .toList();
+  }
+
+  @Override
+  public String getAbsolutePath(Path file) {
+    return file.toAbsolutePath().toString();
+  }
+
+  @Override
+  public String getName(Path file) {
+    return file.getFileName().toString();
+  }
+
+  @Override
+  public InputStream newInputStream(Path file) throws IOException {
+    return Files.newInputStream(file);
+  }
+
+  @Override
+  public OutputStream newOutputStream(Path file) throws IOException {
+    return Files.newOutputStream(file);
+  }
+
+  @Override
+  public BufferedWriter newBufferedWriter(Path file) throws IOException {
+    return Files.newBufferedWriter(file);
+  }
+
+  @Override
+  public BufferedReader newBufferedReader(Path file) throws IOException {
+    return Files.newBufferedReader(file);
+  }
+
+  @Override
+  public Stream<Path> list(Path file) throws IOException {
+    return Files.list(file);
+  }
+
+  @Override
+  public Stream<Path> walk(Path file) throws IOException {
+    return Files.walk(file);
+  }
+
+  @Override
+  public boolean isDirectory(Path file) {
+    return Files.isDirectory(file);
+  }
+
+  @Override
+  public boolean isRegularFile(Path file) {
+    return Files.isRegularFile(file);
+  }
+
+  @Override
+  public FileTime getCreationTime(Path file) throws IOException {
+    var attrs = Files.readAttributes(file, BasicFileAttributes.class);
+    return attrs.creationTime();
+  }
+
+  String listAllFiles() throws IOException {
+    var sb = new StringBuilder();
+    var printFileVisitor = new PrintFileVisitor(sb);
+    Files.walkFileTree(inMemoryRoot, printFileVisitor);
+    return sb.toString();
+  }
+
+  public URI getUri(String path) {
+    var p = inMemoryRoot.resolve(path);
+    return p.toUri();
+  }
+
+  private static final class PrintFileVisitor extends SimpleFileVisitor<Path> {
+    private int depth;
+    private StringBuilder sb;
+
+    private PrintFileVisitor(StringBuilder sb) {
+      this.sb = sb;
+    }
+
+    @Override
+    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+      addEntry(fileName(file));
+      return FileVisitResult.CONTINUE;
+    }
+
+    @Override
+    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+        throws IOException {
+      addEntry(fileName(dir) + "/");
```

Comment:
OS-dependent?

Reply:
This is inside `PrintFileVisitor` that is just a helper for tests that prints the content of the entire virtual file system if a test fails.